src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,482 @@
+                input
+            }).
+            map(|input|
+                Tensor::of_slice(&(input))).
+            collect::<Vec<_>>();
+        Tensor::stack(tokenized_input_tensors.as_slice(), 0).to(self.var_store.device())
+    }
+
+    /// Extract entities from a text
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - `&[&str]` Array of texts to extract entities from.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<Entity>` containing extracted entities
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    ///# fn main() -> failure::Fallible<()> {
+    ///# use rust_bert::pipelines::token_classification::TokenClassificationModel;
+    ///
+    /// let ner_model =  TokenClassificationModel::new(Default::default())?;
+    /// let input = [
+    ///     "My name is Amy. I live in Paris.",
+    ///     "Paris is a city in France."
+    /// ];
+    /// let output = ner_model.predict(&input, true);
+    ///# Ok(())
+    ///# }
+    /// ```
+    pub fn predict(&self, input: &[&str], ignore_first_label: bool) -> Vec<Token> {
+        let input_tensor = self.prepare_for_model(input.to_vec());
+        let (output, _, _) = no_grad(|| {
+            self.token_sequence_classifier
+                .forward_t(Some(input_tensor.copy()),
+                           None,
+                           None,
+                           None,
+                           None,
+                           false)
+        });
+        let output = output.detach().to(Device::Cpu);
+        let score: Tensor = output.exp() / output.exp().sum1(&[-1], true, Float);
+        let labels_idx = &score.argmax(-1, true);
+
+        let mut tokens: Vec<Token> = vec!();
+        for sentence_idx in 0..labels_idx.size()[0] {
+            let labels = labels_idx.get(sentence_idx);
+            let mut word_idx: u8 = 0;
+            for position_idx in 0..labels.size()[0] {
+                let label_id = labels.int64_value(&[position_idx]);
+                let token = {
+                    let token_id = input_tensor.int64_value(&[sentence_idx, position_idx]);
+                    self.decode_token(token_id, label_id, &score, sentence_idx, position_idx, &mut word_idx)
+                };
+                if let Some(token) = token {
+                    if !ignore_first_label || label_id != 0 {
+                        tokens.push(token);
+                    }
+                }
+            }
+        }
+        tokens
+    }
+
+    fn decode_token(&self, token_id: i64, label_id: i64, score: &Tensor, sentence_idx: i64, position_idx: i64, word_idx: &mut u8) -> Option<Token> {
+        let mut text = match self.tokenizer {
+            TokenizerOption::Bert(ref tokenizer) => Tokenizer::decode(tokenizer, vec!(token_id), false, false),
+            TokenizerOption::Roberta(ref tokenizer) => Tokenizer::decode(tokenizer, vec!(token_id), false, false),
+        };
+        let special_value: bool = match self.tokenizer {
```

Comment:
It seems it would be easier to check first if the token_id matches the `unknown_id` of the tokenizer, and then just check against the `special_values` keys. Would make it work the same way for all tokenizers without adding further match branches